File: windowsversion/src/types/project.ts
import type { Tag } from "./common";

/** 项目 — 对应一堂课 / 一个会议 / 一个研究主题 */
export interface Project {
  id: string;
  libraryId: string;
  name: string;
  description: string;
  coverAssetId: string | null;
  source: ProjectSource;
  tags: Tag[];
  isPinned: boolean;
  isArchived: boolean;
  createdAt: string;
  updatedAt: string;
  metadata: ProjectMetadata;
}

/** 项目来源 */
export type ProjectSource =
  | { type: "tf_card"; deviceId: string; sessionId: string }
  | { type: "dropzone" }
  | { type: "manual" };

/** 项目统计元数据 */
export interface ProjectMetadata {
  totalDuration: number | null;
  assetCount: number;
  wordCount: number;
  importedAt: string | null;
}

File: windowsversion/src-tauri/src/models/project.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Project {
    pub id: String,
    pub library_id: String,
    pub name: String,
    pub description: String,
    pub cover_asset_id: Option<String>,
    pub source_type: String,
    pub source_data: Option<String>,
    pub is_pinned: bool,
    pub is_archived: bool,
    pub created_at: String,
    pub updated_at: String,
    pub total_duration: Option<f64>,
    pub asset_count: i64,
    pub word_count: i64,
    pub imported_at: Option<String>,
}
